pkg/client/cache/thread_safe_store.go

```diff
@@ -247,6 +252,16 @@ func (c *threadSafeMap) deleteFromIndices(obj interface{}, key string) error {
 	return nil
 }
 
+//
+func (c *threadSafeMap) Resync() error {
+	c.lock.RLock()
```

Comment:
Actually, this can be empty - if the object doesn't change (which is the case of Resync), updateLocked is effectively a no-op.
So you can simply remove the body of this method.